tests/test_restore_names.py

```diff
@@ -1,3 +1,50 @@
+import pytest
 from app.restore_names import restore_names
 
-# write your tests here
+
+@pytest.mark.parametrize(
+    "users, expected_users",
+    [
+        [
+            [{"first_name": None,
+              "last_name": "Holy",
+              "full_name": "Jack Holy"},
+             {"last_name": "Adams",
+              "full_name": "Mike Adams"}],
+            [{"first_name": "Jack",
+              "last_name": "Holy",
+              "full_name": "Jack Holy"},
+             {"first_name": "Mike",
+              "last_name": "Adams",
+              "full_name": "Mike Adams"}]
+        ]
+    ]
+)
+def test_update_of_none_or_missing_value(users, expected_users):
+    restore_names(users)
+    assert users == expected_users
+
+
+@pytest.mark.parametrize(
+    "users, expected_users",
+    [
+        [
+            [{"first_name": "Jack",
```

Comment:
Same thing.

Reply:
There are many opinions on code styling. As long as flake8 is ok, its ok

Reply:
flake8 is not the only criteria
```
@pytest.mark.parametrize(
    "users, expected_users",
    [
        [
            [
                {"first_name": "Jack", "last_name": "Holy", "full_name": "Jack Holy"},
                {"first_name": "Mike", "last_name": "Adams", "full_name": "Mike Adams"},
            ],
            [
                {"first_name": "Jack", "last_name": "Holy", "full_name": "Jack Holy"},
                {"first_name": "Mike", "last_name": "Adams", "full_name": "Mike Adams"},
            ],
        ]
    ],
)
```
It looks much better